test/unit/SourceLocator/Type/AutoloadSourceLocatorTest.php

```diff
@@ -302,4 +307,63 @@ public function autoload(string $className) : bool
 
         return true;
     }
+
+    /**
+     * @runInSeparateProcess
+     */
+    public function testWillLocateSourcesInPharPath() : void
+    {
+        require_once 'phar://' . __DIR__ . '/../../Fixture/autoload.phar/vendor/autoload.php';
+        spl_autoload_register(static function (string $class) : void {
+            if ($class !== ClassNotInPhar::class) {
+                return;
+            }
+
+            include_once __DIR__ . '/../../Fixture/ClassNotInPhar.php';
+        });
+
+        $sourceLocator  = new AutoloadSourceLocator($this->astLocator);
+        $classReflector = new ClassReflector($sourceLocator);
+
+        $reflection = $classReflector->reflect(AutoloadableClassInPhar::class);
+
+        $this->assertSame(AutoloadableClassInPhar::class, $reflection->getName());
+    }
+
+    public function testBrokenAutoloader() : void
+    {
+        $getErrorHandler = static function () : ?callable {
+            $errorHandler = set_error_handler(static function () : bool {
+                return true;
+            });
+            restore_error_handler();
+
+            return $errorHandler;
+        };
+
+        $toBeThrown           = new BrokenAutoloaderException();
+        $brokenAutoloader     = static function () use ($toBeThrown) : void {
+            throw $toBeThrown;
+        };
+        $previousErrorHandler = $getErrorHandler();
+
+        spl_autoload_register($brokenAutoloader);
+
+        try {
+            (new AutoloadSourceLocator($this->astLocator))
+                ->locateIdentifier(
+                    $this->getMockReflector(),
+                    new Identifier('Whatever', new IdentifierType(IdentifierType::IDENTIFIER_CLASS))
+                );
+
+            self::fail('No exception was thrown');
+        } catch (BrokenAutoloaderException $e) {
+            self::assertSame($e, $toBeThrown);
+        } finally {
+            spl_autoload_unregister($brokenAutoloader);
+        }
+
+        self::assertSame($previousErrorHandler, $getErrorHandler());
+        self::assertTrue(file_exists(__FILE__));
```

Comment:
This will still work even with the replaced stream wrapper: what won't work (if we don't de-register it) is reading the file contents with `file_get_contents()` :D

Reply:
Ok. `file_get_contents` is there :)